src/classes/LookupFieldModel.cls

```diff
@@ -34,9 +34,11 @@
 public with sharing class LookupFieldModel {
     public String name; //developer name
     public String label;
+    public String key;
 
     public LookupFieldModel(String name, String label) {
         this.name = name;
         this.label = label;
+        this.key = name + label;
```

Comment:
The name of a LookupFieldModel is guaranteed to be unique for a given sObjectField. Let's use that instead.